components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java

```diff
@@ -433,21 +459,23 @@ private static void restoreFileStructure(AnalyticsFileSystem analyticsFileSystem
      * @return
      * @throws IOException
      */
-    public static byte[] readFile(File file) throws IOException {
-
-        byte[] buffer = new byte[(int) file.length()];
-        InputStream inputStream = null;
-        try {
-            inputStream = new FileInputStream(file);
-            if (inputStream.read(buffer) == -1) {
-                throw new IOException("EOF reached while trying to read the whole file");
+    private static byte[] readFile(File file) throws IOException {
+
+        byte[] buffer = new byte[READ_BUFFER_SIZE];
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        try (InputStream inputStream = new FileInputStream(file)) {
+            int nRead;
+            while ((nRead = inputStream.read(buffer, 0, (int) file.length())) != -1) {
+                byteArrayOutputStream.write(buffer, 0, nRead);
             }
+        } catch (FileNotFoundException e) {
```

Comment:
shall we move the return statement in to the try block and close the ByteArrayOutputStream in finally block